### PR TITLE
Support for Greek translation and localized messages in Keycloak

### PR DESCRIPTION
**Pull request by infl00p**
Hello,
this Pull Request adds Greek localized messages to the keycloak base theme and keycloak v2 theme's account component.
I'm also working on the translation of the Admin V2 messages which I intent to submit in a separate PR.

It would be great if other Greek-speaking contributors can verify and comment on the messages.

### CODE REVIEW

**Review by jonkoops — COMMENTED** [review]
The changes here look good, but I'd like to see another Greek native speaker review the translated texts.

File: themes/src/main/resources-community/theme/base/email/messages/messages_el.properties [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+emailUpdateConfirmationBodyHtml=<p>Για να ενημερώσετε το λογαριασμό σας {2} με διεύθυνση email {1}, πατήστε το σύνδεσμο παρακάτω </p><p><a href="{0}">{0}</a></p><p>Ο σύνδεσμος θα λήξει σε {3}.</p><p>Αν δεν θέλετε να προχωρήσετε με αυτή την αλλαγή, τότε αγνοήστε αυτό το μήνυμα.</p>
+emailUpdateConfirmationBody=Για να ενημερώσετε το λογαριασμό σας {2} με διεύθυνση email {1}, πατήστε το σύνδεσμο παρακάτω\n\n{0}\n\nΟ σύνδεσμος θα λήξει σε {3}.\n\nΑν δεν θέλετε να προχωρήσετε με αυτή την αλλαγή, τότε αγνοήστε αυτό το μήνυμα.
+emailVerificationBodyCode=Παρακαλώ επιβεβαιώστε τη διεύθυνση email σας, εισάγοντας το παρακάτω κωδικό.\n\n{0}\n\n.
+emailVerificationBodyCodeHtml=<p>Παρακαλώ επιβεβαιώστε τη διεύθυνση email σας, εισάγοντας το παρακάτω κωδικό.</p><p><b>{0}</b></p>
+linkExpirationFormatter.timePeriodUnit.days.1=μέρα
+linkExpirationFormatter.timePeriodUnit.days=μέρες
+linkExpirationFormatter.timePeriodUnit.hours.1=ώρα
+#for language which have more unit plural forms depending on the value (eg. Czech and other Slavic langs) you can override unit text for some other values like this: 
+#linkExpirationFormatter.timePeriodUnit.minutes.2=minuty
```
Comment (EiriniGraonidou) [inline themes/src/main/resources-community/theme/base/email/messages/messages_el.properties:23]: Are those lines (23-25) needed?

File: themes/src/main/resources-community/theme/base/email/messages/messages_el.properties [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+eventUpdateTotpSubject=Ενημέρωση OTP
+eventUpdatePasswordBodyHtml=<p>Ο κωδικός πρόσβασης σας άλλαξε στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.</p>
+eventUpdatePasswordBody=Ο κωδικός πρόσβασης σας άλλαξε στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.
+eventUpdatePasswordSubject=Ενημέρωση κωδικού πρόσβασης
+eventRemoveTotpBodyHtml=<p>Η πρόσβαση μέσω OTP αφαιρέθηκε από το λογαριασμό σας στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.</p>
+eventRemoveTotpBody=Η πρόσβαση μέσω OTP αφαιρέθηκε από το λογαριασμό σας στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.
+eventRemoveTotpSubject=Αφαίρεση OTP
+eventLoginErrorBodyHtml=<p>Ανιχνεύθηκε μία αποτυχημένη προσπάθεια εισόδου στο λογαριασμό σας στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.</p>
+eventLoginErrorBody=Ανιχνεύθηκε μία αποτυχημένη προσπάθεια εισόδου στο λογαριασμό σας στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.
+eventLoginErrorSubject=Σφάλμα εισόδου
```
Comment (EiriniGraonidou) [inline themes/src/main/resources-community/theme/base/email/messages/messages_el.properties:44]:
```suggestion
eventLoginErrorSubject=Σφάλμα Εισόδου
```

File: themes/src/main/resources-community/theme/base/email/messages/messages_el.properties [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+linkExpirationFormatter.timePeriodUnit.seconds.1=δευτερόλεπτο
+
+# units for link expiration timeout formatting
+linkExpirationFormatter.timePeriodUnit.seconds=δευτερόλεπτα
+eventUpdateTotpBodyHtml=<p>Η πρόσβαση μέσω OTP ενημερώθηκε στο λογαριασμό σας στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.</p>
+eventUpdateTotpBody=Η πρόσβαση μέσω OTP ενημερώθηκε στο λογαριασμό σας στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.
+eventUpdateTotpSubject=Ενημέρωση OTP
+eventUpdatePasswordBodyHtml=<p>Ο κωδικός πρόσβασης σας άλλαξε στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.</p>
+eventUpdatePasswordBody=Ο κωδικός πρόσβασης σας άλλαξε στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.
+eventUpdatePasswordSubject=Ενημέρωση κωδικού πρόσβασης
```
Comment (EiriniGraonidou) [inline themes/src/main/resources-community/theme/base/email/messages/messages_el.properties:38]:
```suggestion
eventUpdatePasswordSubject=Ενημέρωση Κωδικού Πρόσβασης
```

File: themes/src/main/resources-community/theme/base/email/messages/messages_el.properties [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+eventLoginErrorBodyHtml=<p>Ανιχνεύθηκε μία αποτυχημένη προσπάθεια εισόδου στο λογαριασμό σας στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.</p>
+eventLoginErrorBody=Ανιχνεύθηκε μία αποτυχημένη προσπάθεια εισόδου στο λογαριασμό σας στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.
+eventLoginErrorSubject=Σφάλμα εισόδου
+executeActionsSubject=Ενημέρωση Του Λογαριασμού Σας
+passwordResetSubject=Αλλαγή κωδικού πρόσβασης
+identityProviderLinkSubject=Σύνδεση {0}
+emailTestBodyHtml=<p>Αυτό είναι ένα δοκιμαστικό μήνυμα</p>
+emailTestBody=Αυτό είναι ένα δοκιμαστικό μήνυμα
+emailTestSubject=[KEYCLOAK] - Δοκιμαστικό μήνυμα SMTP
+emailUpdateConfirmationSubject=Επιβεβαίωση νέου email
```
Comment (EiriniGraonidou) [inline themes/src/main/resources-community/theme/base/email/messages/messages_el.properties:51]:
```suggestion
emailUpdateConfirmationSubject=Επιβεβαίωση Nέου email
```

File: themes/src/main/resources-community/theme/keycloak.v2/account/messages/messages_el.properties [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+signOutWarning=Αποσύνδεση συνεδρίας;
+signedInDevicesExplanation=Αποσύνδεση από όποια άγνωστη συσκευή.
+
+# Device activity page
+signedInDevices=Συνδεδεμένες συσκευές
+
+# Personal info page
+personalInfoHtmlTitle=Προσωπικά στοιχεία
+accountManagementWelcomeMessage=Καλώς ήλθατε στη διαχείριση λογαριασμού στο Keycloak
+backToAdminConsole=Πίσω στη κονσόλα διαχείρισης
```
Comment (EiriniGraonidou) [inline themes/src/main/resources-community/theme/keycloak.v2/account/messages/messages_el.properties:154]:
```suggestion
backToAdminConsole=Πίσω στην κονσόλα διαχείρισης
```

File: themes/src/main/resources-community/theme/base/email/messages/messages_el.properties [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+linkExpirationFormatter.timePeriodUnit.minutes.1=λεπτό
+linkExpirationFormatter.timePeriodUnit.minutes=λεπτά
+linkExpirationFormatter.timePeriodUnit.seconds.1=δευτερόλεπτο
+
+# units for link expiration timeout formatting
+linkExpirationFormatter.timePeriodUnit.seconds=δευτερόλεπτα
+eventUpdateTotpBodyHtml=<p>Η πρόσβαση μέσω OTP ενημερώθηκε στο λογαριασμό σας στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.</p>
+eventUpdateTotpBody=Η πρόσβαση μέσω OTP ενημερώθηκε στο λογαριασμό σας στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.
+eventUpdateTotpSubject=Ενημέρωση OTP
+eventUpdatePasswordBodyHtml=<p>Ο κωδικός πρόσβασης σας άλλαξε στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.</p>
```
Comment (EiriniGraonidou) [inline themes/src/main/resources-community/theme/base/email/messages/messages_el.properties:36]:
```suggestion
eventUpdatePasswordBodyHtml=<p>Ο κωδικός πρόσβασής σας άλλαξε στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.</p>
```

File: themes/src/main/resources-community/theme/base/email/messages/messages_el.properties [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+eventUpdatePasswordBody=Ο κωδικός πρόσβασης σας άλλαξε στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.
```
Comment (EiriniGraonidou) [inline themes/src/main/resources-community/theme/base/email/messages/messages_el.properties:37]:
```suggestion
eventUpdatePasswordBody=Ο κωδικός πρόσβασής σας άλλαξε στις {0} από το {1}. Αν δεν είσαστε εσείς, τότε παρακαλώ επικοινωνήστε με ένα διαχειριστή.
```

File: themes/src/main/resources-community/theme/keycloak.v2/account/messages/messages_el.properties [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+recovery-codes-generate-new-codes=Δημιουργήστε νέους κωδικούς για να βεβαιωθείτε ότι θα έχετε πρόσβαση στο λογαριασμό σας
+recovery-codes-number-remaining={0} κωδικοί ανάκτησης απομένουν
+recovery-codes-number-used={0} κωδικοί ανάκτησης χρησιμοποιήθηκαν
+recovery-authn-codes-help-text=Αυτοί οι κωδικοί μπορούν να χρησιμοποιηθούν για να επανακτήσετε τη πρόσβαση σε περίπτωση που οι άλλες σας μέθοδοι δύο-παραγόντων δεν είναι διαθέσιμες.
+recovery-authn-codes-display-name=Κωδικοί ανάκτησης ταυτοποίησης
+recovery-authn-code=Οι κωδικοί ανάκτησης ταυτοποίησης μου
+otp-help-text=Εισάγετε ένα κωδικό επιβεβαίωσης από την εφαρμογή ταυτοποίησης.
+otp-display-name=εφαρμογή ταυτοποίησης
+password=Ο κωδικός πρόσβασης μου
+password-help-text=Συνδεθείτε βάζοντας το κωδικό πρόσβασης σας.
```
Comment (EiriniGraonidou) [inline themes/src/main/resources-community/theme/keycloak.v2/account/messages/messages_el.properties:77]:
```suggestion
password-help-text=Συνδεθείτε βάζοντας το κωδικό πρόσβασής σας.
```

File: themes/src/main/resources-community/theme/base/admin/messages/messages_el.properties [hunk 9]
```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+
+error-invalid-date=Το χαρακτηριστικό {0} δεν είναι έγκυρη ημερομηνία.
```
[the remark this answers is not in the document]
Comment (EiriniGraonidou) [reply themes/src/main/resources-community/theme/base/admin/messages/messages_el.properties:3]: I would suggest the usage of the word πεδίο as a translation for field/attribute/property.

File: themes/src/main/resources-community/theme/base/admin/messages/messages_el.properties [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+error-invalid-length-too-long=Το χαρακτηριστικό {0} πρέπει να έχει μέγιστο μήκος {2}.
```
[the remark this answers is not in the document]
Comment (EiriniGraonidou) [reply themes/src/main/resources-community/theme/base/admin/messages/messages_el.properties:4]: Although the usage of "μέγιστο μήκος" (maximum length) has been adopted in the technical vocabulary, I still find it strange as a Greek native speaker. Thus the above suggestion. The initial translation is valid, though.

**Review by EiriniGraonidou — COMMENTED** [review]
Good job, the translation seems very legit! :sunglasses: I have only noticed some typos that could be fixed. :nerd_face: 

I also proposed some alternatives for the usage of some words (pretty much opionion based).

**Comment by jonkoops** [comment]
Thanks for hopping on this @EiriniGraonidou, much appreciated!

**Comment by infl00p** [comment]
Hi @EiriniGraonidou ,
thank you for your suggestions !

**Comment by jonkoops** [comment]
@EiriniGraonidou Thanks for providing a review, can I ask you to take another look to confirm the messages now look ok?

File: themes/src/main/resources-community/theme/base/admin/messages/messages_el.properties [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+error-invalid-length-too-long=Το πεδίο {0} μπορεί να περιέχει το πολύ {2} χαρακτήρες .
+error-person-name-invalid-character=Το {0} περιέχει ένα μη έγκυρο χαρακτήρα.
+error-username-invalid-character=Το {0} περιέχει ένα μη έγκυρο χαρακτήρα.
+error-user-attribute-read-only=Το χαρακτηριστικό {0} είναι για ανάγνωση μόνο.
+error-user-attribute-required=Παρακαλώ ορίστε το χαρακτηριστικό {0}.
+error-invalid-uri-fragment=Μη έγκυρο κομμάτι URL.
+error-invalid-uri-scheme=Μη έγκυρο σχήμα URL.
+error-invalid-uri=Μη έγκυρο URL.
+error-pattern-no-match=Μη έγκυρη τιμή.
+error-number-out-of-range-too-big=Το χαρακτηριστικό {0} πρέπει να έχει μέγιστη τιμή {2}.
```
Comment (EiriniGraonidou) [inline themes/src/main/resources-community/theme/base/admin/messages/messages_el.properties:13]:
```suggestion
error-number-out-of-range-too-big=Το πεδίο {0} μπορεί να έχει μέγιστη τιμή {2}.
```

File: themes/src/main/resources-community/theme/base/admin/messages/messages_el.properties [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+error-number-out-of-range-too-small=Το χαρακτηριστικό {0} πρέπει να έχει ελάχιστη τιμή {1}.
```
Comment (EiriniGraonidou) [inline themes/src/main/resources-community/theme/base/admin/messages/messages_el.properties:14]:
```suggestion
error-number-out-of-range-too-small=Το πεδίο {0} μπορεί να έχει ελάχιστη τιμή {1}.
```

File: themes/src/main/resources-community/theme/base/admin/messages/messages_el.properties [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+ldapErrorCantWriteOnlyForReadOnlyLdap=Αδυναμία ορισμού σε εγγραφή-μόνο, όταν ο πάροχος LDAP δεν είναι σε ΕΓΓΡΑΨΙΜΗ λειτουργία
+ldapErrorCantPreserveGroupInheritanceWithUIDMembershipType=Δεν είναι δυνατή η ταυτόχρονη διατήρηση της κληρονομιάς ομάδας και χρήση του τύπου μελών UID.
+ldapErrorMissingClientId=Το Client ID πρέπει να παρέχεται στις ρυθμίσεις όταν δε χρησιμοποιείται η Αντιστοίχηση Ρόλων Τομέα.
+ldapErrorReadTimeoutNotNumber=Η Λήξη Χρόνου Ανάγνωσης πρέπει να είναι αριθμός
+ldapErrorConnectionTimeoutNotNumber=Η Λήξη Χρόνου Σύνδεσης πρέπει να είναι αριθμός
+ldapErrorInvalidCustomFilter=Το προσαρμοσμένο φίλτρο LDAP που ορίστηκε δεν ξεκινά με "(" ή δε τελειώνει με ")".
+ldapErrorEditModeMandatory=Η Λειτουργία Επεξεργασίας είναι υποχρεωτική
+invalidPasswordGenericMessage=Μη έγκυρος κωδικός πρόσβασης: ο νέος κωδικός δε συμφωνεί με τις πολιτικές κωδικών.
+invalidPasswordBlacklistedMessage=Μη έγκυρος κωδικός πρόσβασης: ο κωδικός είναι απαγορευμένος.
+invalidPasswordHistoryMessage=Μη έγκυρος κωδικός πρόσβασης: δε πρέπει να είναι το ίδιο με τους τελευταίους {0} κωδικούς.
```
Comment (EiriniGraonidou) [inline themes/src/main/resources-community/theme/base/admin/messages/messages_el.properties:56]:
```suggestion
invalidPasswordHistoryMessage=Μη έγκυρος κωδικός πρόσβασης: δεν επιτρέπεται να είναι ο ίδιος με τους τελευταίους {0} κωδικούς.
```

File: themes/src/main/resources-community/theme/base/admin/messages/messages_el.properties [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+invalidPasswordRegexPatternMessage=Μη έγκυρος κωδικός πρόσβασης: δε ταιριάζει με τα μοτίβα regex.
```
Comment (EiriniGraonidou) [inline themes/src/main/resources-community/theme/base/admin/messages/messages_el.properties:57]:
```suggestion
invalidPasswordRegexPatternMessage=Μη έγκυρος κωδικός πρόσβασης: δεν ταιριάζει με τα μοτίβα regex.
```

**Review by EiriniGraonidou — COMMENTED** [review]
All typos have been fixed!

My previous comment suggested the translation of the field/attribute to be πεδίο.

For sake of consistency I either suggest to apply this translation universally (see the new suggestions) or rollback the previous ones (using the word χαρακτηριστικό)

As  already mentioned though both work fine, and my comment was pretty much subjective.

Cheers, and sorry for any misunderstandings..

**Comment by jonkoops** [comment]
Thanks again @EiriniGraonidou! @infl00p could you process the latest feedback?

**Comment by jonkoops** [comment]
@infl00p there are still some comments from @EiriniGraonidou that have not been addressed. Could you resolve these as well? You can see them by pressing the following link:

image

File: themes/src/main/resources-community/theme/base/email/messages/messages_el.properties (hunk 1 above)
Comment (infl00p) [reply themes/src/main/resources-community/theme/base/email/messages/messages_el.properties:23]: These are comments they can ommited ofcourse

**Comment by infl00p** [comment]
@jonkoops  Sorry for the delay I am on summer vacations and do not read mails.

**Review by jonkoops — APPROVED** [review]
LGTM!

**Comment by jonkoops** [comment]
No problem! We're really appreciative of you taking the time to contribute this. I am going to go ahead and merge in the work as it is now, as it's already a great enhancement! If there are any translations in need of improvement, feel free to open up a new PR in the future.

Thanks for your contributions @infl00p @EiriniGraonidou!

**Review by ssilvert — APPROVED** [review]

**Pull request merged**